record-and-playback/core/lib/recordandplayback/edl/video.rb

```
@@ -394,6 +446,207 @@ def self.render(edl, layout, output_basename, threads = 2)
+
+        BigBlueButton.logger.debug "Processing segments for '#{File.basename(filename)}': #{segments.inspect}"
+        segments
+      end
+
+      # Resamples video files by identifying all large PTS gaps, cutting at surrounding keyframes,
+      # re-encoding gappy segments (by replacing with blank video), and stream-copying stable segments.
+      def self.resample_video_files(files_and_gap_info_map, target_directory, videoinfo_h)
+        filename_update_map = {}
+        intermediate_format = 'webm'
+
+        files_and_gap_info_map.each do |original_filename, _first_gap_info_ignored| # First gap info is ignored, we find all gaps
```

RuboCop (1.82.1)

[convention] Line is too long. [129/128]

(Layout/LineLength)